src/gsi/general_read_gfsatm.f90

```diff
@@ -3588,99 +3588,99 @@ subroutine general_read_gfsatm_allhydro_nc(grd,sp_a,filename,uvflag,vordivflag,z
           if ( icount == icm .or. k==nlevs) then
              call general_reload2(grd,g_z,g_ps,g_tv,g_vor,g_div,g_u,g_v,g_q,g_oz, &
                   g_ql,g_qi,g_qr,g_qs,g_qg,icount,iflag,ilev,work,uvflag,vordivflag,g_ni,g_nr)
-      endif
-   enddo ! do k=1,nlevs
-
-   do k=1,nlevs
-      icount=icount+1
-      iflag(icount)=15
-      ilev(icount)=k
-      kr = levs+1-k ! netcdf is top to bottom, need to flip
+          endif
+       enddo ! do k=1,nlevs
 
-      if (mype==mype_use(icount)) then
-         call read_vardata(filges, 'nccice', rwork3d0, nslice=kr, slicedim=3)
-         ! cloud ice water number concentration.
-         if ( diff_res ) then
-            grid_b=rwork3d0(:,:,1)
-            vector(1)=.false.
-            call fill2_ns(grid_b,grid_c(:,:,1),latb+2,lonb)
-            call g_egrid2agrid(p_high,grid_c,grid2,1,1,vector)
-            do kk=1,grd%itotsub
-               i=grd%ltosi_s(kk)
-               j=grd%ltosj_s(kk)
-               work(kk)=grid2(i,j,1)
-            enddo
-         else
-            grid=rwork3d0(:,:,1)
-            call general_fill_ns(grd,grid,work)
-         endif
-      endif
-      if ( icount == icm .or. k==nlevs ) then
-         call general_reload2(grd,g_z,g_ps,g_tv,g_vor,g_div,g_u,g_v,g_q,g_oz, &
-              g_ql,g_qi,g_qr,g_qs,g_qg,icount,iflag,ilev,work,uvflag,vordivflag,g_ni,g_nr)
-      endif
-   enddo ! do k=1,nlevs
+       if (imp_physics == 8) then
```

Comment:
Suggestion:  Add a comment noting that this section is for processing model backgrounds generated using Thompson microphysics.   Not all developers know / remember what `imp_physics=8` represents.

Reply:
@emilyhcliu , what do you think about the suggestion to add a comment on [line 3594](https://github.com/emilyhcliu/GSI/blob/3404ddfef03fd52c1fa04411d6613e15b7ef77da/src/gsi/general_read_gfsatm.f90#L3594) explaining what `imp_physics=8` represents.   For example,
```
       enddo ! do k=1,nlevs

#      Read fields specific to Thompson microphysics
       if (imp_physics == 8) then   
          do k=1,nlevs
```

Reply:
That's good and should be done.  I will add that.

Reply:
Thank you @emilyhcliu